server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java

```diff
@@ -90,16 +88,12 @@ protected ShardIterator shards(ClusterState state, InternalRequest request) {
         if (iterator == null) {
             return null;
         }
-        // If it is stateless, only route promotable shards. This is a temporary workaround until a more cohesive solution can be
+        // Only route to promotable shards. This is a temporary workaround for stateless until a more cohesive solution can be
         // implemented for search shards.
-        if (DiscoveryNode.isStateless(clusterService.getSettings())) {
-            return new PlainShardIterator(
-                iterator.shardId(),
-                iterator.getShardRoutings().stream().filter(ShardRouting::isPromotableToPrimary).collect(Collectors.toList())
-            );
-        } else {
-            return iterator;
-        }
+        return new PlainShardIterator(
+            iterator.shardId(),
+            iterator.getShardRoutings().stream().filter(ShardRouting::isPromotableToPrimary).toList()
+        );
```

Comment:
This change looks unrelated and also looks like something we will touch anyway in the mget? Perhaps we can leave this out?

Reply:
done in https://github.com/elastic/elasticsearch/pull/93976/commits/101a09f601bc57269d28f0276636cff22a73b02d